src/shells/webextension/contentScript.js

```diff
@@ -12,7 +12,17 @@ import debugConnection from '../../utils/debugConnection';
 const contentScriptId = Math.random().toString(32).slice(2);
 
 // proxy from main page to devtools (via the background page)
-const port = chrome.runtime.connect({ name: 'content-script' });
+let port = chrome.runtime.connect({ name: 'content-script' });
+
+// Handle port disconnection and reconnection
+port.onDisconnect.addListener(() => {
+  // Try to reconnect after a short delay
+  setTimeout(() => {
+    const newPort = chrome.runtime.connect({ name: 'content-script' });
+    // Update port reference
+    port = newPort;
```

Comment:
Maybe! I am not 100% sure how all of this is wired up. What should we be doing on port disconnection?

Reply:
Not sure, I just see that you try reconnecting after the first disconnection here, thought it would make sense to do the same for the next disconnections